### PR TITLE
[MAINTENANCE] Clean up additional deprecation warnings from outdated CRUD API

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- More cleanup!

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

**Comment by ghost** [comment]
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=7156&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

#### 

[Review these changes using an interactive CodeSee Map](https://app.codesee.io/r/reviews?pr=7156&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
CodeSee Map Legend
</details>

File: great_expectations/cli/v012/toolkit.py [hunk 1]
```diff
@@ -295,7 +295,7 @@ def create_empty_suite(
     )
     suite = context.create_expectation_suite(expectation_suite_name)
     suite.add_citation(comment="New suite added via CLI", batch_kwargs=batch_kwargs)
-    context.save_expectation_suite(suite, expectation_suite_name)
```
Comment (NathanFarmer) [inline great_expectations/cli/v012/toolkit.py:298]: What is the difference between `save_expectation_suite` and `update_expectation_suite`?

File: tests/actions/test_validation_operators_in_data_context.py [hunk 2]
```diff
@@ -55,12 +55,11 @@ def validation_operators_data_context(
...
-        failure_expectations, expectation_suite_name="f1.failure"
-    )
-    data_context.save_expectation_suite(
-        warning_expectations, expectation_suite_name="f1.warning"
-    )
+    failure_expectations.expectation_suite_name = "f1.failure"
+    data_context.add_expectation_suite(expectation_suite=failure_expectations)
+
+    warning_expectations.expectation_suite_name = "f1.warning"
+    data_context.save_expectation_suite(expectation_suite=warning_expectations)
```
Comment (NathanFarmer) [inline tests/actions/test_validation_operators_in_data_context.py]: What is the difference between `save_expectation_suite` and `update_expectation_suite`?

File: tests/cli/test_suite.py [hunk 3]
```diff
@@ -894,15 +894,15 @@ def test_suite_new_interactive_valid_batch_request_from_json_file_in_notebook_ru
...
-        ),
+        # mock.call(
+        #     {
+        #         "event": "data_context.save_expectation_suite",
+        #         "event_payload": {
+        #             "anonymized_expectation_suite_name": "9df638a13b727807e51b13ec1839bcbe"
+        #         },
+        #         "success": True,
+        #     }
+        # ),
```
Comment (NathanFarmer) [inline tests/cli/test_suite.py]: Why not remove this?

File: tests/cli/test_suite.py [hunk 4]
```diff
@@ -1379,15 +1379,15 @@ def test_suite_edit_multiple_datasources_with_no_additional_args_without_citatio
...
```
Comment (NathanFarmer) [inline tests/cli/test_suite.py]: Same question for the rest of these.

File: tests/integration/profiling/rule_based_profiler/test_profiler_basic_workflows.py [hunk 5]
```diff
@@ -326,7 +326,7 @@ def test_profiler_save_and_load(data_context_with_taxi_data):
                         "_amount",
                     ],
                 },
-                "variables": {},
```
Comment (NathanFarmer) [inline tests/integration/profiling/rule_based_profiler/test_profiler_basic_workflows.py]: I wonder what caused this?

File: great_expectations/cli/v012/toolkit.py (hunk 1 above)
Comment (cdkini) [reply great_expectations/cli/v012/toolkit.py:298]: `save_expectation_suite` is an old method that essentially works as `add` (can overwrite existing). `update` takes in the same arguments but can only update an existing object. This is to be consistent across our API:
- `add` - only new objects
- `update` - only existing objects
- `add_or_update` - essentially upsert

File: tests/actions/test_validation_operators_in_data_context.py (hunk 2 above)
Comment (cdkini) [reply tests/actions/test_validation_operators_in_data_context.py]: whoops this should also be `add` - same answer as above though

File: tests/cli/test_suite.py (hunk 3 above)
Comment (cdkini) [reply tests/cli/test_suite.py]: fixed

File: tests/cli/test_suite.py (hunk 4 above)
Comment (cdkini) [reply tests/cli/test_suite.py]: fixed

File: tests/integration/profiling/rule_based_profiler/test_profiler_basic_workflows.py (hunk 5 above)
Comment (cdkini) [reply tests/integration/profiling/rule_based_profiler/test_profiler_basic_workflows.py]: hmm I'm trying to dig into it some more - seems weird

File: tests/integration/docusaurus/expectations/advanced/user_configurable_profiler_cross_table_comparison_from_query.py [hunk 6]
```diff
@@ -112,9 +112,8 @@
 # <snippet name="tests/integration/docusaurus/expectations/advanced/user_configurable_profiler_cross_table_comparison_from_query.py build_suite">
 expectation_suite_name = "compare_two_tables"
 suite = profiler.build_suite()
-context.save_expectation_suite(
-    expectation_suite=suite, expectation_suite_name=expectation_suite_name
-)
+suite.expectation_suite_name = expectation_suite_name
+context.save_expectation_suite(expectation_suite=suite)
```
Comment (NathanFarmer) [inline tests/integration/docusaurus/expectations/advanced/user_configurable_profiler_cross_table_comparison_from_query.py]: `save`

File: tests/integration/docusaurus/expectations/advanced/user_configurable_profiler_cross_table_comparison.py [hunk 7]
```diff
@@ -106,9 +106,8 @@
 # <snippet name="tests/integration/docusaurus/expectations/advanced/user_configurable_profiler_cross_table_comparison.py build_suite">
 expectation_suite_name = "compare_two_tables"
 suite = profiler.build_suite()
-context.save_expectation_suite(
-    expectation_suite=suite, expectation_suite_name=expectation_suite_name
-)
+suite.expectation_suite_name = expectation_suite_name
+context.save_expectation_suite(expectation_suite=suite)
```
Comment (NathanFarmer) [inline tests/integration/docusaurus/expectations/advanced/user_configurable_profiler_cross_table_comparison.py]: `save`

File: great_expectations/cli/v012/toolkit.py (hunk 1 above)
Comment (NathanFarmer) [reply great_expectations/cli/v012/toolkit.py:298]: So why are we using `save_expectation_suite` still?

Comment (cdkini) [reply great_expectations/cli/v012/toolkit.py:298]: whoops you're right!

**Review by NathanFarmer — APPROVED** [review]

**Pull request merged**